ee/server/service/service_teams.go

```diff
@@ -286,3 +287,81 @@ func (svc *Service) ModifyTeamEnrollSecrets(ctx context.Context, teamID uint, se
 
 	return newSecrets, nil
 }
+
+func (svc Service) ApplyTeamSpecs(ctx context.Context, specs []*fleet.TeamSpec) error {
+	if err := svc.authz.Authorize(ctx, &fleet.Team{}, fleet.ActionRead); err != nil {
+		return err
+	}
+
+	// check auth for all teams specified first
+	for _, spec := range specs {
+		team, err := svc.ds.TeamByName(ctx, spec.Name)
+		if err != nil {
+			if err := ctxerr.Cause(err); err == sql.ErrNoRows {
+				// can the user create a new team?
+				if err := svc.authz.Authorize(ctx, team, fleet.ActionWrite); err != nil {
```

Comment:
I wouldn't rely on the result of a method if `err != nil`, so just use `&fleet.Team{}`.